pages/flashbar/interactive.page.tsx

```diff
@@ -3,34 +3,60 @@
 import * as React from 'react';
 import { useState } from 'react';
 import { range } from 'lodash';
-import { Button, SpaceBetween, Flashbar, FlashbarProps, Toggle } from '~components';
+import { Box, Button, SpaceBetween, Flashbar, FlashbarProps, Toggle } from '~components';
 import { generateItem, i18nStrings } from './common';
+import ScreenshotArea from '../utils/screenshot-area';
 
 export default function InteractiveFlashbar() {
   const dismiss = (index: string) => {
     setItems(items => items.filter(item => item.id !== index));
   };
 
   const add = (type: FlashbarProps.Type, hasHeader = false) => {
-    setItems(items => [generateItem(type, dismiss, hasHeader), ...items]);
+    setNextId(id => {
+      setItems(items => [generateItem({ type, id: id.toString(), dismiss, hasHeader }), ...items]);
+      return id + 1;
+    });
```

Comment:
I might need an opinion from someone with a lil' more React knowledge but this looks a bit odd. Is it fine to update state from within here?

Reply:
This was to make sure `items` are updated using the current `nextId`. In other words, to make the state update operation more atomic.

Otherwise I was running into a situation where, if for example `add` was being called from another function, a JS closure was being created that led to calling `add` with an old value of `nextId`.

Reply:
Here is an example using this approach: https://stackoverflow.com/a/62806870

Reply:
Here is the page (thanks for the deploy feature!!) if you want to interact with it: https://d21d5uik3ws71m.cloudfront.net/components/aed5a5b9f5cedae2a51004d37b3958864b45f10a/index.html#/light/flashbar/interactive

Reply:
It seems a more conventional approach, with two updates, also works fine as long as I don't wrap both calls in a `setTimeout`:

```
setItems(items => [generateItem({ type, id: nextId.toString(), dismiss, hasHeader }), ...items]);
setNextId(id => id + 1);
```

I can change that if you think it's more straightforward/conventional, I have no strong preference.

Reply:
`nextId` never used in rendering, it is only a counter. You should use ref for such values.

This way you will stay away from possible undocumented behaviors

Reply:
Thanks, this works well. Updated